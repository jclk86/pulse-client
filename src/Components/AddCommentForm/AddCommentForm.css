.AddCommentForm {
  max-width: 800px;
  margin-bottom: 100px;
  margin-top: 50px;
  margin-left: auto;
  margin-right: auto;
}

.label_add_comment_form {
  display: block;
  width: 90%;
  margin: 5px auto;
  text-align: left;
}

.AddCommentForm_comment_box {
  width: 90%;
  padding: 50px;
  resize: none;
  border-radius: 10px;
  text-align: center;
  font-size: 16px;
}

.AddCommentForm_comment_box:focus {
  outline-width: 0;
}

.AddCommentForm_submit_btn {
  display: block;
  margin: 10px auto;
  border-radius: 10px;
  min-width: 200px;
  padding: 6px;
  font-size: 16px;
}

.AddCommentForm_submit_btn:hover {
  cursor: pointer;
}

.AddCommentForm_submit_btn:focus {
  outline-width: 0;
}
