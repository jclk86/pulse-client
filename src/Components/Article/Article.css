.container_article_image {
  width: 100%;
  margin: 30px auto;
}

.article_image {
  width: 100%;
  object-fit: cover;
  height: 300px;
}

.container_article {
  max-width: 800px;
  margin: 100px auto;
  padding-bottom: 50px;
  padding-right: 0;
}

.link_user_profile {
  text-decoration: none;
  color: rgb(51, 160, 255);
  font-weight: bold;
}

.link_user_profile:hover {
  text-decoration: underline;
}

.article_date,
.article_author {
  font-size: 16px;
}

.article_content {
  white-space: pre-line;
  text-align: justify;
  text-indent: 5%;
}

.Article_edit_post_btn {
  min-width: 200px;
  border-radius: 10px;
  font-size: 16px;
  border: none;
  background-color: rgb(90, 90, 90);
  padding: 10px;
  color: white;
  font-weight: bold;
  background-color: rgb(51, 160, 255);
}

.Article_edit_post_btn:hover {
  cursor: pointer;
}

.Article_edit_post_btn:focus {
  outline-width: 0;
}

@media only screen and (min-width: 568px) {
  .container_article_image {
    max-width: 800px;
  }
  .article_image {
    height: 400px;
    object-fit: cover;
  }
}
