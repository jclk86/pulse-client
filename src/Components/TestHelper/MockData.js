import { DateFormatter } from "../TestHelper/MockData";

export const mockUser = {
  username: "Pop123",
  password: "Password123!",
  fullname: "Bob Smith",
  email: "BSmith@gmail.com"
};

export const mockArticle = {
  title: "test article",
  content: "test content",
  article_tag: "News",
  article_id: 1,
  date_created: new Date(),
  author: {
    username: "Pop123"
  }
};
