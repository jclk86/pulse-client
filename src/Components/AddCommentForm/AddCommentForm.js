import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import ArticleContext from "../../Context/ArticleContext";
import CommentService from "../../Services/comment-api-service";
import PropTypes from "prop-types";
import "./AddCommentForm.css";

class AddCommentForm extends Component {
  static contextType = ArticleContext;
  constructor(props) {
    super(props);
    this.state = {
      article_id: this.props.article_id,
      comment: { value: "", touched: false }
    };
  }

  updateComment = comment => {
    this.setState({ comment: { value: comment, touched: true } });
  };

  handleSubmit = event => {
    event.preventDefault();
    const { comment, article_id } = this.state;
    const newComment = {
      content: comment.value,
      article_id
    };

    CommentService.postComment(newComment)
      .then(this.context.addComment)
      .then(() => {
        this.setState({ comment: { value: "", touched: false } });
        this.props.history.push(`/articles/${article_id}`);
      });
  };
  render() {
    return (
      <form
        className="AddCommentForm"
        onSubmit={event => this.handleSubmit(event)}
      >
        <label
          htmlFor="AddCommentForm_comment_title"
          className="label_add_comment_form"
        >
          Add Comment:
        </label>
        <textarea
          type="text"
          placeholder="Tell us your thoughts!"
          name="comment"
          className="AddCommentForm_comment_box"
          onChange={e => this.updateComment(e.target.value)}
          value={this.state.comment.value}
        ></textarea>
        <button type="submit" className="AddCommentForm_submit_btn">
          Submit
        </button>
      </form>
    );
  }
}

AddCommentForm.propTypes = {
  article_id: PropTypes.string
};

export default withRouter(AddCommentForm);
