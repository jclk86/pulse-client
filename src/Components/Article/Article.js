import React, { Component } from "react";
import { withRouter, NavLink } from "react-router-dom";
import ArticleContext from "../../Context/ArticleContext";
import TokenService from "../../Services/token-service";
import { Button } from "../Utils/Utils";
import { DateFormatter } from "../Utils/Utils";
import PropTypes from "prop-types";
import "./Article.css";

class Article extends Component {
  static contextType = ArticleContext;

  render() {
    const { article } = this.props;
    const token = TokenService.readJwtToken();
    console.log(token.user_id);
    console.log(article.date_created);
    return (
      <div className="container_article">
        <div className="container_image">IMAGE SECTION</div>
        <h2>{article.title}</h2>
        <p>{DateFormatter(article.date_created)}</p>
        <p>
          <span>By</span> {article.author.username}
        </p>
        <p>{article.content}</p>
        {token.user_id === article.author.id ? (
          <NavLink to={`/articles/${article.id}/edit_article`}>
            <Button type="button">Edit Post</Button>
          </NavLink>
        ) : (
          <div></div>
        )}
      </div>
    );
  }
}

Article.propTypes = {
  article: PropTypes.shape({
    title: PropTypes.string,
    content: PropTypes.string,
    date_created: PropTypes.string
  })
};

export default withRouter(Article);
