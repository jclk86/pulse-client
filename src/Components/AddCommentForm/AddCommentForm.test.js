import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import sinon from "sinon";
import AddCommentForm from "./AddCommentForm";
import Article from "../../Components/Article/Article";
import AuthApiService from "../../Services/auth-api-service";
import ArticleApiService from "../../Services/article-api-service";
import { mockUser, mockArticle } from "../TestHelper/MockData";

describe(`AddCommentForm Component`, () => {
  beforeAll(() => {
    AuthApiService.postUser(mockUser).then(user => {
      AuthApiService.postLogin({
        username: user.username,
        password: user.password
      });
    });

    ArticleApiService.postArticle(mockArticle);
  });

  it(`renders AddCommentForm without issues`, () => {
    const wrapper = shallow(<AddCommentForm></AddCommentForm>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("submits AddCommentForm", () => {
    const mockComment = {
      content: "test comment content",
      article_id: 1
    };
    const onSubmit = sinon.spy();
    const wrapper = mount(
      <BrowserRouter>
        <AddCommentForm onSubmit={onSubmit}></AddCommentForm>
      </BrowserRouter>
    );

    wrapper.find(".AddCommentForm").simulate("submit", {
      target: {
        content: mockComment.content,
        article_id: mockComment.article_id
      },
      preventDefault: () => {}
    });
  });
});
