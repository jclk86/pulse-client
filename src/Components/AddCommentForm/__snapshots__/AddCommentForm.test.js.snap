// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCommentForm Component renders AddCommentForm without issues 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
